### DISCUSSION TITLE
Cant load/validate on some URL arg

### DISCUSSION THREAD

**Discussion by lawrencenika** [developmentseed/titiler#984] [Q&A]
#### Problem description

Tried on some COG path in GCS worked, but this one is not working: `http://localhost:8000/cog/validate?url=https://storage.googleapis.com/raster-datasources/Hansen%20GFC%202023%20v1.11%2000N%20050W.tif`

then I got
```
{
    "detail": "'/vsicurl/https://storage.googleapis.com/raster-datasources/Hansen GFC 2023 v1.11 00N 050W.tif' not recognized as a supported file format."
}
```

when I tried on url `http://localhost:8000/cog/validate?url=gs://abcd/325dc85d-5903-4477-becd-cf48184a0745/Hansen GFC 2023 v1.11 00N 050W.tif` it returns

```
{
    "detail": "'/vsigs/planetgpt/325dc85d-5903-4477-becd-cf48184a0745/Hansen GFC 2023 v1.11 00N 050W.tif' does not exist in the file system, and is not recognized as a supported dataset name."
}
```

but it worked on `http://localhost:8000/cog/validate?url=gs://abcd/325dc85d-5903-4477-becd-cf48184a0745/Hansen GFC 2023 v1.11 lossyear 00N 050W.tif`

#### Expected Output

JSON output

#### Environment Information

titiler:latest (v0.18.6) run in local docker env on macos
with ENV WORKERS_PER_CORE 1

**Comment by vincentsarago** [comment]
Dear @lawrencenika 
This can have plenty of reason:
- malformed url
- you don't have the right to access the file
- GDAL config is not correctly set to have access to Google bucket
- ..... 

I would also avoid to have ` ` space in my file name

**Reply by lawrencenika** [reply]
thanks for the quick response, for the 2 examples I gave, the only difference is an extra `lossyear` text in the 2 difference objects in GCS bucket. i dont get why that is malformed. can you direct me to the code i can look into and debug

**Reply by vincentsarago** [reply]
can you try to do `gdalinfo /vsigs/abcd/325dc85d-5903-4477-becd-cf48184a0745/Hansen GFC 2023 v1.11 lossyear 00N 050W.tif` command 


https://gdal.org/en/latest/user/virtual_file_systems.html#vsigs-google-cloud-storage-files

**Reply by lawrencenika** [reply]
let me try here, the problem is that my data is in private bucket, so in titiler I set the authenticated token access, if I use `gdalinfo` locally i m not sure if I can achieve the same

**Comment by lawrencenika** [comment]
false alarm, my bad, turns out my url encode was not consistent and it could work for the test file.
